apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx

```diff
@@ -116,6 +132,42 @@ export default function WeaponSwapModal({
     [dbDirty, database, rarity, weaponTypeKey, deferredSearchTerm]
   )
 
+  const { weaponIdsToShow, numPages, currentPageIndex } = useMemo(() => {
```

Comment:
Looks like we could make this a util to be shared between arti/weapon/char pagination?

Reply:
I'm also ok just saving that for a different PR

Reply:
Lets do another PR later, I am moving some code around this area and would be nice to have less noise.